### PR TITLE
[ FEAT ] : Add Login and Signup Page

### LINKED ISSUE #198: [ FEAT ] : Add Login and Signup Page
### Description

This feature introduces fully functional Login and Signup pages, leveraging modern UI components, React Hook Form, and Zod for seamless form handling and robust validation.

### Features

**Login Page:**

- User authentication with email and password fields.
- Real-time validation and error handling.
- Responsive design for an optimal user experience on all devices.

**Signup Page:**

- User registration with email, password, and confirm password fields.
- Client-side validation using Zod to ensure data integrity.
- Integration with React Hook Form for efficient form state management.

### PR DESCRIPTION
**Pull request by senthil-athiban**
**What does this PR do?**

This PR closes #198, which aim is to enhance the Auth Page.

**What issues does this PR fix or reference?**

- Leverage React-hook-form for form management
- Adding Zod for client side validations of the input
- Enhancing the UI by integrating modern UI and re-usable components

### ScreenRecorder

https://github.com/user-attachments/assets/1efd2ee5-3c74-4dbd-ae0b-73dcbe07d912

----------------------------------------------------------------------------------------------------------------------

### Screen Shot

**Signup Page**

Screenshot 2024-07-30 at 12 50 32 PM
----------------------------------------------------------------------------------------------------------------------

**Login Page**

Screenshot 2024-07-30 at 12 56 24 PM

### CODE REVIEW

File: frontend/src/pages/Login.tsx [hunk 1]
```diff
@@ -51,16 +59,21 @@ const Login = () => {
...
-          <Form onSubmit={handleGithubSubmit}>
-            <TCButton type="submit">
-              <FontAwesomeIcon icon={faGithub} style={{ marginRight: 15 }} />
-              Login with Github
-            </TCButton>
-          </Form>
+          <div className={styles.container}>
+            <Card className="custom-card">
+              <CardHeader>
+                <CardTitle>Welcome to KScaleLabs</CardTitle>
```
Comment (codekansas) [inline frontend/src/pages/Login.tsx:65]: nit - can change `KScaleLabs` to `Robolist`

File: frontend/src/pages/Login.tsx [hunk 2]
```diff
@@ -51,16 +59,21 @@ const Login = () => {
...
+                <CardDescription>Login</CardDescription>
+              </CardHeader>
+              <CardContent className={styles.content}>
+                <Input placeholder="Email" type="text" />
+                <Input placeholder="Password" type="password" />
+              </CardContent>
```
Comment (codekansas) [inline frontend/src/pages/Login.tsx:71]: we don't use email + password authentication right now? maybe i'm not following what this is for

**Review by codekansas — CHANGES_REQUESTED** [review]
maybe i'm not following what is going on here? could you add a screenshot of the page after making ui changes?

**Comment by senthil-athiban** [comment]
Sure @codekansas

**Comment by senthil-athiban** [comment]
@codekansas can we use tailwind css library, instead of using seperate css module?

File: frontend/src/pages/Login.tsx (hunk 2 above)
Comment (senthil-athiban) [reply frontend/src/pages/Login.tsx:71]: Adding credentials (email, password) + providers (github, Google) auth

**Comment by codekansas** [comment]
yes, tailwind is fine, if we can make it work well with react typescript - cc @chennisden

**Comment by chennisden** [comment]
> yes, tailwind is fine, if we can make it work well with react typescript - cc @chennisden

yeah i've done this before, tailwind plays perfectly fine with react ts

**Comment by senthil-athiban** [comment]
@codekansas, I've added credentials + google + github provider in the Client. If we are currently supporting github provider means, we can disable the rest of the things, in future we may need this auth credentials. Your thoughts on these ?

Screenshot 2024-07-30 at 12 50 32 PM

**Comment by codekansas** [comment]
this looks great!! let's merge this

**Review by codekansas — APPROVED** [review]
this is incredible, nice

**Pull request merged**